src/reconcile/rec_write.c

```diff
@@ -2484,25 +2484,28 @@ __rec_las_wrapup(WT_SESSION_IMPL *session, WT_RECONCILE *r)
  * __rec_las_wrapup_err --
  *     Discard any saved updates from the database's lookaside buffer.
  */
-static int
+static void
 __rec_las_wrapup_err(WT_SESSION_IMPL *session, WT_RECONCILE *r)
 {
-    WT_DECL_RET;
+    WT_BIRTHMARK_DETAILS *birthmarkp;
     WT_MULTI *multi;
-    uint64_t las_pageid;
-    uint32_t i;
+    uint32_t i, j;
 
     /*
-     * Note the additional check for a non-zero lookaside page ID, that flags if lookaside table
-     * entries for this page have been written.
+     * Note the additional check for whether lookaside table entries for this page have been
+     * written.
      */
     for (multi = r->multi, i = 0; i < r->multi_next; ++multi, ++i)
-        if (multi->supd != NULL && (las_pageid = multi->page_las.las_pageid) != 0) {
-            WT_TRET(__wt_las_remove_block(session, las_pageid));
+        if (multi->supd != NULL && multi->page_las.has_las) {
+            if (multi->page_las.birthmarks_cnt != 0) {
+                WT_ASSERT(session, multi->page_las.birthmarks != NULL);
+                for (j = 0; j < multi->page_las.birthmarks_cnt; j++) {
+                    birthmarkp = multi->page_las.birthmarks + j;
+                    __wt_buf_free(session, &birthmarkp->key);
+                }
+            }
```

Comment:
Still not a fan of `birthmarkp = multi->page_las.birthmarks + j;`

I guess we're consistent. :)